app/src/main/res/layout/activity_main.xml

```diff
@@ -10,125 +10,183 @@
         android:id="@+id/screen_text_example"
         style="@style/calculatorScreen"
         android:layout_width="match_parent"
-        android:layout_height="wrap_content" />
-    
-    <GridLayout
+        android:layout_height="100dp" />
+
+    <LinearLayout
         android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:columnCount="@integer/grid_layout_column_count"
-        android:rowCount="@integer/grid_layout_row_count">
+        android:layout_height="match_parent"
+        android:layout_weight="25"
+        android:orientation="horizontal"
+        android:weightSum="100">
 
         <Button
             android:id="@+id/button_number_seven"
             style="@style/calculatorButton"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_margin="5dp"
+            android:layout_weight="25"
             android:text="@string/main_screen_btn_text_seven" />
 
         <Button
             android:id="@+id/button_number_eight"
             style="@style/calculatorButton"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_margin="5dp"
+            android:layout_weight="25"
             android:text="@string/main_screen_btn_text_eight" />
 
         <Button
             android:id="@+id/button_number_nine"
             style="@style/calculatorButton"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_margin="5dp"
+            android:layout_weight="25"
             android:text="@string/main_screen_btn_text_nine" />
 
         <Button
             android:id="@+id/button_function_sum"
-            style="@style/calculatorButton"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
+            style="@style/calculatorOperationsButtons"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_margin="1dp"
+            android:layout_weight="25"
```

Comment:
why do you use this property?

Reply:
At first it helped me to to divide the screen equitably, but then thanks to your explanation I understood that it is better to use the weight property in each of the children, so that it is the father who is in charge of dividing and performing the pertinent calculation.